anxcloud/data_source_dns_records.go

```diff
@@ -0,0 +1,34 @@
+package anxcloud
+
+import (
+	"context"
+	"github.com/anexia-it/go-anxcloud/pkg/client"
+	"github.com/anexia-it/go-anxcloud/pkg/clouddns/zone"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func dataSourceDnsRecords() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataSourceDNSRecordsRead,
+		Schema:      schemaDNSRecords(),
+	}
+}
+
+func dataSourceDNSRecordsRead(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	c := m.(client.Client)
+	a := zone.NewAPI(c)
+
+	zoneName := d.Get("zone_name").(string)
```

Comment:
This being a string is checked by terraform? If not, this will panic when not a string

Reply:
so from you changing this ... did you check if terraform already checks the type? I don't know enough about terraform, so it's possible this is useless nitpicking to have this changed.

Checking anyway will not cause problems, but more than needed verbose code can be a (slight) maintenance burden.

Reply:
thanks for checking :)